README.md

```diff
@@ -1,12 +1,36 @@
 # Swarm Stack Updater
-The Swarm Stack Updater an automated tool built in shell that can be run periodically on a docker swarm to search for and identify any out of date stacks. If any changes are found, this tool automatically downloads, updates, and redeploys software back onto the swarm.
+The Swarm Stack Updater an automated tool built in shell that can be run periodically on a Docker Swarm to search for and identify any out of date stacks. If any changes are found, this tool automatically downloads, updates, and redeploys software back onto the swarm.
 
-This tool has been developed in conjuction with the University of Canterbury Computer Science Education Research Group as a University of Canterbury Sofware Engineering final year project. 
+This tool has been specifically enginneered to find and update full stacks by downloading Docker Compose files from GitHub. This was after finding tools such as [Watchtower](https://github.com/containrrr/watchtower) or [Shepard](https://github.com/djmaze/shepherd) which only aimed to update individual Docker containers by updating images. Using Compose files means whole application stacks can be updated and can allow for configuration changes as well.  
 
-## How to use:
 
-### Deploying Using Docker
-* First create a config file matching this description:
+## Requirements
+* Needs to be run in a docker swarm, this can either be locally or on a remove server.
+* The tool makes use of the Github API and Container Registry. You may have to update the script to work for other 
+repositories and container registries.
+* Website must have a status url that returns a JSON object that contains both 
+sha commit hash and a tag version.
+* ```{"VERSION_NUMBER": "3.10.0", "GIT_SHA": "a49a111d"}```
+* Deployment on swarm requires Swarm Cronjob running and deployed on swarm
+this can be found here https://github.com/crazy-max/swarm-cronjob
+* From the bottom of the compose file you can see that a secret is required. This should be a ```github_access_token``` to allow for multiple requests. More about docker secrets can be found here: https://docs.docker.com/engine/swarm/secrets/
+
+* Your docker images should also contain some extra information in their manifest files. This is so the app is able to identify whether a new image is available to download.
+
+Manifest JSON should contain at least:
+```
+...
+labels: {
+  "org.opencontainers.image.revision": "<commit-sha>"
+}
+...
+```
+
+## Deployment:
+
+1. First create a config file so that the Swarm Stack Updater is able to recognise what it needs to update. This should contain the stack's name, whether you are deploying the tool in a development environment, the websites url that you are updating as well as some repository information.
```

Comment:
```suggestion
1. First create a config file so that the Swarm Stack Updater is able to recognise what it needs to update.
  This should contain the stack's name, whether you are deploying the tool in a development environment, the websites url that you are updating as well as some repository information.
```